inmem/kv.go

```diff
@@ -50,9 +60,18 @@ func (s *KVStore) View(ctx context.Context, fn func(kv.Tx) error) error {
 
 // Update opens up a transaction with a write lock.
 func (s *KVStore) Update(ctx context.Context, fn func(kv.Tx) error) error {
-	s.mu.Lock()
+	if !s.mu.TryLock(ctx) {
+		return ctx.Err()
+	}
 	defer s.mu.Unlock()
 
+	// we have to do this because RTryLock doesn't check context before attempting to take the lock
+	select {
```

Comment:
@docmerlin why this?

It seems that `TryLock` accounts for `ctx.Done()` in the select statement 🤔

Reply:
Woops, that was my fault, thanks affo!